src/renderer/widgets/WalletDetails/model/wallet-connect-details-model.ts

```diff
@@ -0,0 +1,73 @@
+import { combine, createEvent, createStore, forward, sample } from 'effector';
+
+import { accountUtils } from '@renderer/entities/wallet';
+import { InitConnectProps, walletConnectModel } from '@renderer/entities/walletConnect';
+import { ReconnectStep } from '../common/const';
+import { walletProviderModel } from './wallet-provider-model';
+
+const reset = createEvent();
+const reconnectStarted = createEvent<Omit<InitConnectProps, 'client'>>();
+const reconnectAborted = createEvent();
+const sessionTopicUpdated = createEvent();
+
+const $reconnectStep = createStore<ReconnectStep>(ReconnectStep.NOT_STARTED).reset(reset);
+
+const $connected = combine(walletProviderModel.$accounts, walletConnectModel.$client, (accounts, client): boolean => {
+  const account = accounts[0];
+  if (!client || !account || !accountUtils.isWalletConnectAccount(account)) return false;
+
+  const sessions = client?.session.getAll() || [];
+
+  const storedSession = sessions.find((s) => s.topic === accounts[0].signingExtras?.sessionTopic);
+
+  return Boolean(storedSession);
+});
+
+sample({
+  clock: reconnectStarted,
+  fn: () => ReconnectStep.RECONNECTING,
+  target: $reconnectStep,
+});
+
+forward({
+  from: reconnectStarted,
+  to: walletConnectModel.events.connect,
+});
+
+sample({
+  clock: walletConnectModel.events.connected,
+  source: {
+    accounts: walletProviderModel.$accounts,
+    session: walletConnectModel.$session,
+  },
+
+  filter: ({ accounts, session }) => accounts.length > 0 && Boolean(session?.topic),
+  fn: ({ accounts, session }) => ({
```

Comment:
I guess this should work
```suggestion
fn: ({ accounts, session }) => ({ accounts, topic: session!.topic }),
```